lib/screens/home/view_more/hidden_proposal_creation/hidden_proposal_creation_screen.dart

```diff
@@ -0,0 +1,274 @@
+                        ),
+                        PwListDivider.alternate(),
+                        DetailsItem.withRowChildren(
+                          title: strings.devInitialDepositAmount,
+                          children: [
+                            PwText(
+                              strings.hashAmount(
+                                  details.initialDeposit.toString()),
+                              style: PwTextStyle.footnote,
+                            )
+                          ],
+                        ),
+                        DepositSlider(
+                          max: details.hashAmount.toDouble(),
+                          thumbColor: Theme.of(context).colorScheme.primary550,
+                          onChanged: (changed) =>
+                              _bloc.updateInitialDeposit(changed),
+                        ),
+                        PwListDivider.alternate(),
+                        PwGasAdjustmentSlider(
+                          title: strings.stakingConfirmGasAdjustment,
+                          startingValue: defaultGasEstimate,
+                          onValueChanged: (value) {
+                            setState(() {
+                              _gasEstimate = value;
+                            });
+                          },
+                        ),
+                      ],
+                    ),
+                  ),
+                  PwListDivider.alternate(),
+                  VerticalSpacer.large(),
+                  Padding(
+                    padding: EdgeInsets.only(
+                      left: Spacing.large,
+                      right: Spacing.large,
+                      bottom: Spacing.largeX3,
+                    ),
+                    child: PwButton(
+                      enabled: details.title.isNotEmpty &&
+                          details.description.isNotEmpty,
+                      onPressed: () async {
+                        await ModalLoadingRoute.showLoading(
+                          context,
+                          minDisplayTime: Duration(milliseconds: 500),
+                        );
+                        await _sendProposal(_gasEstimate, context);
+                      },
+                      child: PwText(
+                        strings.devConfirmProposal,
+                        softWrap: false,
+                        overflow: TextOverflow.fade,
+                        color: PwColor.neutralNeutral,
+                        style: PwTextStyle.body,
+                      ),
+                    ),
+                  ),
+                ],
+              );
+            }),
+      ),
+    );
+  }
+
+  Future<void> _sendProposal(
+    double? gasEstimate,
+    BuildContext context,
+  ) async {
+    final navigator = Navigator.of(context);
+    try {
+      final response = await _bloc.sendTransaction(gasEstimate);
+      ModalLoadingRoute.dismiss(context);
```

Comment:
Probably want this in a `finally` block

Reply:
This one is a weird one because the modal dismissal needs to happen immediately and if I push it to a finally, then all that other logic needs to be pushed as well.

The modal dismissal is already in the catch block.

Reply:
Okay, follow up question: if I put it in a `finally`, is it weird if navigation happens but the modal is still up for a second?

Reply:
Hmm. It seems like maybe the code that shows the`ModalLoadingRoute` should also be the code that dismisses the`ModalLoadingRoute`.

Reply:
That's a good point. Let me take a look at it.